### ISSUE TITLE
Auto Update Tags in Files Plugin saves before tag actually gets updated

### ISSUE THREAD

**Issue by h88e22dgpeps56sg** [quodlibet/quodlibet#3717]
Steps to reproduce
------------------
- Activate Plugin: _Auto Update Tags in Files_
- Update strategy: Either of the two containing: After every play
- Play a song until the end and wait for the next song to start or skip after more than half of the song was played
- Inspect the last played song with an external ID3 tag editor

Expected Output
---------------
- Quodlibet updates the tag – most visibly playcount
- Plugin saves updated tag to file

Actual Output
-------------
- Plugin saves "old" tag to file
- Quodlibet updates the tag – most visibly playcount

Test System
-----------

### Which version of Quod Libet?
4.4

### Which operating system 
Arch Linux

Additional Information
----------------------
Not much, sadly. From debug output I can see where the tag gets updated from QL. But I cannot see when the plugin operates.

**Comment by CreamyCookie** [comment]
When I wrote this plugin I thought that plugins are called by Quod Libet after the playcount got increased, and I feel like at the time, when I was testing the plugin, the playcount was indeed increased before saving.

Anyway, maybe the playcount increase could be moved before the plugin call.. Might be hard, and it's possible other plugins rely on the way it works now. We might be able to hotfix it by doing this in the AutoUpdateTagsInFiles plugin:

0. Plugin is called by QL with song
1. Increase its playcount by one
2. Write tags to its file.
3. Decrease playcount again.

**Comment by h88e22dgpeps56sg** [comment]
Thanks for your reply.

Correct me if I'm wrong, but that would lead to the playcount tag in files being incremented even if it won't actually get incremented afterwards, e. g. because it was skipped early.

In this case I would prefer the state as is: Less complex and just temporally off, not actually off.

**Comment by CreamyCookie** [comment]
@h88e22dgpeps56sg 

You're right, but it's possible to check whether a track was skipped as the signature is

```python
    def plugin_on_song_ended(self, song, skipped):
```

**Comment by h88e22dgpeps56sg** [comment]
That was just a quick example. The real problem is to find out wheter the playcount got incremented or not. You cannot infer this from the fact that the song was skipped (or not).

**Comment by cidthecoatrack** [comment]
I know this one is old now, but I'd like to bump it. I have a network drive to sync my music between multiple machines. I use the Auto-Update to make sure play counts and ratings match between different machines/instances of QL. Right now, since auto-update tags doesn't actually save the updated/incremented play count, I have to go to my "recently played", select al lthe tracks I've played that day, and manually update their tags to make sure that the playcount propagates correctly. It's a real pain in the butt, and it would be much better if the plugin could update the playcount tag when it does its auto-update.
